src/gui/editors/BBEditor.cpp

```diff
@@ -264,3 +261,27 @@ void BBTrackContainerView::updatePosition()
 	//realignTracks();
 	emit positionChanged( m_currentPosition );
 }
+
+
+
+
+void BBTrackContainerView::makeSteps( bool clone )
+{
+	TrackContainer::TrackList tl = model()->tracks();
+
+	for( TrackContainer::TrackList::iterator it = tl.begin();
+		it != tl.end(); ++it )
+	{
+		if( ( *it )->type() == Track::InstrumentTrack )
+		{
+			Pattern* p = static_cast<Pattern *>( ( *it )->getTCO( m_bbtc->currentBB() ) );
+			if( clone )
```

Comment:
Don't you think this code would be easier to understand if it was 2 methods, one for clone and one for add?